Migrate SSL Channel FAT

isaacrivriv opened the pull request:
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [ ] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [ ] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

Adds SSL Channel FAT testing to Open Liberty

isaacrivriv committed:
Added first steps to migrate ssl channel options test to liberty

isaacrivriv committed:
First working migrated tests
- Added necessary logic for tests to build and run
- Added first two tests and confirmed passing
- Added keys for SSL negotiation
- Removed unused items from initial commit

isaacrivriv committed:
Added dynamic update test and log cap test for handshake failures

isaacrivriv committed:
Added all tests and updated to work with repeat action and multiple servlet versions

isaacrivriv committed:
Updated unused trace mark to remove unnecessary trace

isaacrivriv commented:
#build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=3cb166af-94b4-413d-b238-058925649eb4) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

pnicolucci commented on dev/io.openliberty.transport.ssl_fat/bnd.bnd:
```diff
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
```
If you can remove the contributors from these classes / files that would be a good clean up

pnicolucci commented on dev/io.openliberty.transport.ssl_fat/fat/src/io/openliberty/transport/ssl/FATSuite.java:
```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2024 IBM Corporation and others.
```
Should we have an update to 2025 here since we're adding a repeat or is this exactly as it was in ws-cd-open?

isaacrivriv replied:
I made some change so I'll update most of these to 2025

pnicolucci commented on dev/io.openliberty.transport.ssl_fat/fat/src/io/openliberty/transport/ssl/SSLOptionsTest.java:
```diff
@@ -0,0 +1,341 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2023 IBM Corporation and others.
```
I won't comment on every one but if there are any significant changes to any of the tests we should update the range to include 2025

pnicolucci commented on dev/io.openliberty.transport.ssl_fat/publish/files/SSLOptions.xml:
```diff
@@ -0,0 +1,53 @@
+ -->
+<server description="FAT for channel SSL - alternate SSLConfig">
+
+    <include optional="true" location="../fatTestPorts.xml"/>
```
Should we just use fatTestCommon.xml rather than the fatTestPorts.xml since we're defining the endpoint in these server.xml and then add the portOpenRetries so we don't lose that from fatTestPorts.xml?

I believe if you look at the logs we're going to see messages about duplicate endpoints or something like that similar to what I saw / fixed in : https://github.com/OpenLiberty/open-liberty/issues/31081

isaacrivriv replied:
I used pretty much the same xml configuration as in the tests in ws-cd-open which came with the fatTestPorts xml but I can update to fatTestCommon xml especially if we've seen issues before with it

pnicolucci commented on dev/io.openliberty.transport.ssl_fat/fat/src/io/openliberty/transport/ssl/SSLOptionsTest.java:
```diff
@@ -0,0 +1,341 @@
+     */
+    @Test
+    public void handshakeFailureGetsLogged() throws Exception {
+        LOG.info("Entering handshakeFailureGetsLogged");
```
I'm not seeing a server.xml in the SSLOptionsServer only a bootstrap.properties so what server.xml are we using for this test? I seem to be missing where we are setting the default server.xml

pnicolucci replied:
It would help I guess if I scrolled down....resolving

isaacrivriv committed:
Added servlet-3.0 on tested features bnd

isaacrivriv committed:
Updated copyrights in files for cleanup

isaacrivriv commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=f723dc42-8006-4f28-8a34-1ffdaff432c7) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

isaacrivriv committed:
Fixed changes for build errors
- Added allowed FFDC for IllegalArgument exception unrelated to tests
- Added ignore error on server shutdown
- Added SSL endpoint wait on configuration restore

isaacrivriv commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=19819c18-f7d2-49db-a204-5cc1b783fff9) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

isaacrivriv committed:
Corrected wait for string to use log mark

isaacrivriv committed:
Clean up comments unrelated to the tests

isaacrivriv commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=503ebfc8-69de-4939-b238-e90c742946d9) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

pnicolucci commented on dev/io.openliberty.transport.ssl_fat/bnd.bnd:
```diff
@@ -0,0 +1,51 @@
+
+# To define a global minimum java level for the FAT, use the following property.
+# If unspecified, the default value is ${javac.source}
+# fat.minimum.java.level: 11
```
We should just clean this up since it is not necessary

isaacrivriv committed:
Requested clean up of commented lines from creation of bucket

isaacrivriv commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=a397e584-af0a-4d22-9a9f-bdea24b66004) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

isaacrivriv committed:
Added additional wait for strings on certain occasions due to some failures

isaacrivriv commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=2cd5f113-e6e0-4926-9ba6-1944f3d6e21a) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._
